src/main/java/mcbot/McBot.java

```diff
@@ -0,0 +1,236 @@
+                        Task t;
+                        if (taskName.isBlank()) {
+                            throw new InvalidCommandException("you can't leave your event task empty");
+                        }
+                        String dateStr = parser.getEventDate();
+                        LocalDate eventDate = LocalDate.parse(dateStr, dateFormatter);
+                        if (dateStr.isBlank()) {
+                            throw new InvalidCommandException("you can't leave your event date/time empty");
+                        }
+                        if (parser.isThereTime()) {
+                            String timeStr = parser.getEventTime();
+                            LocalTime eventTime = LocalTime.parse(timeStr, timeFormatter);
+                            t = new Event(taskName, eventDate, eventTime);
+                        } else {
+                            t = new Event(taskName, eventDate);
+                        }
+                        tasks.add(t);
+                        storage.appendData(t);
+                        ui.addEventLine(t, tasks.size());
+                    } catch (InvalidCommandException e) {
+                        ui.printError(e);
+                    } catch (ArrayIndexOutOfBoundsException e) {
+                        ui.taskError("eventFormat");
+                    } catch (DateTimeParseException e) {
+                        ui.taskError("datetimeFormat");
+                    }
+                    break;
+                case "delete":
+                    try {
+                        String numStr = parser.getDetails();
+                        int num = Integer.parseInt(numStr);
+                        if (num < 1 || num > tasks.size()) {
+                            throw new McBotException("Boi, I can't delete a number that ain't on the list");
+                        }
+                        Task t = tasks.get(num - 1);
+                        tasks.remove(num - 1);
+                        storage.updateData(tasks.getList());
+                        ui.deleteLine(t, tasks.size());
+                    } catch (InvalidCommandException | ArrayIndexOutOfBoundsException e) {
+                        ui.deleteError("empty");
+                    } catch (McBotException e) {
+                        ui.printError(e);
+                    } catch (NumberFormatException e) {
+                        ui.deleteError("notInteger");
+                    }
+                    break;
+                case "find":
+                    try {
+                        String taskName = parser.getDetails();
+                        tasks.find(taskName, ui);
+                    } catch (InvalidCommandException | ArrayIndexOutOfBoundsException e) {
+                        ui.taskError("emptyFindTask");
+                    }
+                    break;
+                default:
+                    throw new InvalidCommandException("I don't understand a word ye're sayin'");
```

Comment:
——— For PR review ———
Might I suggest more usage of OOP-styles -- create a command object for each command that stores its actions and the aliases that call forth it.